test/framework/autoscaler_helpers.go

```diff
@@ -0,0 +1,426 @@
+	patchHelper, err := patch.NewHelper(input.Cluster, mgmtClient)
+	Expect(err).ToNot(HaveOccurred())
+	for i := range input.Cluster.Spec.Topology.Workers.MachineDeployments {
+		md := input.Cluster.Spec.Topology.Workers.MachineDeployments[i]
+		if md.Metadata.Annotations == nil {
+			md.Metadata.Annotations = map[string]string{}
+		}
+		// Add the autoscaler annotation
+		md.Metadata.Annotations[clusterv1.AutoscalerMinSizeAnnotation] = input.NodeGroupMinSize
+		md.Metadata.Annotations[clusterv1.AutoscalerMaxSizeAnnotation] = input.NodeGroupMaxSize
+		// Drop the replicas from MachineDeploymentTopology, or else the topology controller and autoscaler with fight over control.
+		md.Replicas = nil
+		input.Cluster.Spec.Topology.Workers.MachineDeployments[i] = md
+	}
+	Eventually(func(g Gomega) {
+		g.Expect(patchHelper.Patch(ctx, input.Cluster)).Should(Succeed())
+	}, retryableOperationTimeout, retryableOperationInterval).Should(Succeed(), "Failed to patch Cluster topology to add autoscaler annotations")
+
+	log.Logf("Wait for the annotations to applied on the MachineDeployments")
+	Eventually(func(g Gomega) {
+		mdList := GetMachineDeploymentsByCluster(ctx, GetMachineDeploymentsByClusterInput{
+			Lister:      mgmtClient,
+			ClusterName: input.Cluster.Name,
+			Namespace:   input.Cluster.Namespace,
+		})
+		for i := range mdList {
+			md := mdList[i]
+			g.Expect(md.Annotations).To(HaveKey(clusterv1.AutoscalerMinSizeAnnotation))
+			g.Expect(md.Annotations).To(HaveKey(clusterv1.AutoscalerMaxSizeAnnotation))
+		}
+	}, input.WaitForAnnotationsToBeAdded...).Should(Succeed(), "Auto scaler annotations are missing from the MachineDeployments")
+}
+
+// getAuthenticationTokenForAutoscaler returns a bearer authenticationToken with minimal RBAC permissions that will be used
+// by the autoscaler running on the workload cluster to access the management cluster.
+func getAuthenticationTokenForAutoscaler(ctx context.Context, managementClusterProxy ClusterProxy, namespace string, cluster string, infraMachineTemplateKind string) string {
+	name := fmt.Sprintf("cluster-%s", cluster)
+	sa := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	Expect(managementClusterProxy.GetClient().Create(ctx, sa)).To(Succeed(), fmt.Sprintf("failed to create %s service account", name))
+
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:     []string{"get", "list", "update", "watch"},
+				APIGroups: []string{"cluster.x-k8s.io"},
+				Resources: []string{"machinedeployments", "machinedeployments/scale", "machines", "machinesets"},
+			},
+			{
+				Verbs:     []string{"get", "list"},
+				APIGroups: []string{"infrastructure.cluster.x-k8s.io"},
+				Resources: []string{strings.ToLower(infraMachineTemplateKind)},
```

Comment:
I would prefer just documenting that the `infraMachineTemplateKind` folks are setting at the spec is already lower case.

Upper case plural what they can set today just doesn't make a lot of sense